### DISCUSSION TITLE
AsyncQueue

### DISCUSSION THREAD

**Discussion by john-livelyvideo** [esfx/esfx#38] [Ideas]
@rbuckton  Just curious.  Any idea how fast AsyncQueue should be, once warmed up, when compared to c# channels?

The reason for asking.  For very large in memory data structures, where there will often be cache line misses, I'm wondering if multiple AsyncQueue's can be used to ensure the cpu pipeline is kept loaded (via out of order execution) w/ multiple cache line loads?  The intention, we're basically trying to "await" cache line misses and page faults w/ heavy use of Promise.any and numerous AsyncQueue's.

**Comment by rbuckton** [comment]
I haven't reached the point of fine-tuning performance for `AsyncQueue`, so I doubt its anywhere near as fast as what you'd need for something like that.
